### ISSUE TITLE
Azure installation error: TLS match must have at least one SNI host

### ISSUE THREAD

**Issue by uipo78** [kubeflow/manifests#1667]
I'm attempting to deploy kubeflow on an AKS v1.18.8 cluster, but encountered the following error: 

```
WARN[0055] Encountered error applying application istio:  (kubeflow.error): Code 500 with message: Apply.Run : [error when applying patch:
{"spec":{"tls":[{"match":[{"port":443,"sni_hosts":["www.googleapis.com"]}],"route":[{"destination":{"host":"www.googleapis.com","port":{"number":443}},"weight":100}]}]}}
to:
Resource: "networking.istio.io/v1alpha3, Resource=virtualservices", GroupVersionKind: "networking.istio.io/v1alpha3, Kind=VirtualService"
Name: "google-api-vs", Namespace: "kubeflow"
Object: &{map["apiVersion":"networking.istio.io/v1alpha3" "kind":"VirtualService" "metadata":map["annotations":map["kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"networking.istio.io/v1alpha3\",\"kind\":\"VirtualService\",\"metadata\":{\"annotations\":{},\"name\":\"google-api-vs\",\"namespace\":\"kubeflow\"},\"spec\":{\"hosts\":[\"www.googleapis.com\"],\"tls\":[{\"match\":[{\"port\":443,\"sni_hosts\":[\"www.googleapis.com\"]}],\"route\":[{\"destination\":{\"host\":\"www.googleapis.com\",\"port\":{\"number\":443}},\"weight\":100}]}]}}\n"] "creationTimestamp":"2020-11-16T23:31:52Z" "generation":'\x01' "managedFields":[map["apiVersion":"networking.istio.io/v1alpha3" "fieldsType":"FieldsV1" "fieldsV1":map["f:metadata":map["f:annotations":map[".":map[] "f:kubectl.kubernetes.io/last-applied-configuration":map[]]] "f:spec":map[".":map[] "f:hosts":map[] "f:tls":map[]]] "manager":"kfctl" "operation":"Update" "time":"2020-11-16T23:31:52Z"]] "name":"google-api-vs" "namespace":"kubeflow" "resourceVersion":"7078778" "selfLink":"/apis/networking.istio.io/v1alpha3/namespaces/kubeflow/virtualservices/google-api-vs" "uid":"2a2b4c93-f128-414a-a383-9b031540fd98"] "spec":map["hosts":["www.googleapis.com"] "tls":[map["match":[map["port":'\u01bb']] "route":[map["destination":map["host":"www.googleapis.com" "port":map["number":'\u01bb']] "weight":'d']]]]]]}
for: "/tmp/kout247395750": admission webhook "pilot.validation.istio.io" denied the request: configuration is invalid: TLS match must have at least one SNI host, error when applying patch:
{"spec":{"tls":[{"match":[{"port":443,"sni_hosts":["storage.googleapis.com"]}],"route":[{"destination":{"host":"storage.googleapis.com","port":{"number":443}},"weight":100}]}]}}
to:
Resource: "networking.istio.io/v1alpha3, Resource=virtualservices", GroupVersionKind: "networking.istio.io/v1alpha3, Kind=VirtualService"
Name: "google-storage-api-vs", Namespace: "kubeflow"
Object: &{map["apiVersion":"networking.istio.io/v1alpha3" "kind":"VirtualService" "metadata":map["annotations":map["kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"networking.istio.io/v1alpha3\",\"kind\":\"VirtualService\",\"metadata\":{\"annotations\":{},\"name\":\"google-storage-api-vs\",\"namespace\":\"kubeflow\"},\"spec\":{\"hosts\":[\"storage.googleapis.com\"],\"tls\":[{\"match\":[{\"port\":443,\"sni_hosts\":[\"storage.googleapis.com\"]}],\"route\":[{\"destination\":{\"host\":\"storage.googleapis.com\",\"port\":{\"number\":443}},\"weight\":100}]}]}}\n"] "creationTimestamp":"2020-11-16T23:31:52Z" "generation":'\x01' "managedFields":[map["apiVersion":"networking.istio.io/v1alpha3" "fieldsType":"FieldsV1" "fieldsV1":map["f:metadata":map["f:annotations":map[".":map[] "f:kubectl.kubernetes.io/last-applied-configuration":map[]]] "f:spec":map[".":map[] "f:hosts":map[] "f:tls":map[]]] "manager":"kfctl" "operation":"Update" "time":"2020-11-16T23:31:52Z"]] "name":"google-storage-api-vs" "namespace":"kubeflow" "resourceVersion":"7078779" "selfLink":"/apis/networking.istio.io/v1alpha3/namespaces/kubeflow/virtualservices/google-storage-api-vs" "uid":"096e5753-267d-4509-953e-a6da22c515df"] "spec":map["hosts":["storage.googleapis.com"] "tls":[map["match":[map["port":'\u01bb']] "route":[map["destination":map["host":"storage.googleapis.com" "port":map["number":'\u01bb']] "weight":'d']]]]]]}
for: "/tmp/kout247395750": admission webhook "pilot.validation.istio.io" denied the request: configuration is invalid: TLS match must have at least one SNI host]  filename="kustomize/kustomize.go:266"
```

This is the KfDef file that I `kfctl apply -f`'ed:

```yaml
apiVersion: kfdef.apps.kubeflow.org/v1
kind: KfDef
metadata:
  namespace: kubeflow
spec:
  applications:  
  - kustomizeConfig:
      repoRef:
        name: manifests
        path: namespaces/base
    name: namespaces
  - kustomizeConfig:
      repoRef:
        name: manifests
        path: application/v3
    name: application
  - kustomizeConfig:
      repoRef:
        name: manifests
        path: stacks/azure/application/istio-1-3-1-stack
    name: istio-stack
  - kustomizeConfig: 
      repoRef:
        name: manifests
        path: stacks/kubernetes/application/cluster-local-gateway-1-3-1
    name: cluster-local-gateway
  - kustomizeConfig:
      repoRef:
        name: manifests
        path: stacks/azure/application/istio
    name: istio 
  - kustomizeConfig:
      repoRef:
        name: manifests
        path: metacontroller/base
    name: metacontroller
  - kustomizeConfig:
      repoRef:
        name: manifests
        path: stacks/azure/application/oidc-authservice
    name: oidc-authservice
  - kustomizeConfig:
      repoRef:
        name: manifests
        path: stacks/azure
    name: kubeflow-apps
  repos:
  - name: manifests
    uri: https://github.com/kubeflow/manifests/archive/v1.2-branch.tar.gz    
  version: v1.2-branch
```

Note that I removed the cert-manager components since it'd blow away the cluster's pre-existing cert-manager instance.

Not sure where I should post this issue (here or in the main `kubeflow` repo), so let me know if I should move it elsewhere. Thanks!

**Comment by berndverst** [comment]
Was Istio preinstalled already? If so, which version? The AAD / multi user integration right now must deploy the bundled Istio 1.3.1. All distributions of Kubeflow have very strong ties to specific older Istio versions for multi-tenancy at this time.

By the way, download the v1.2 manifest again because I had to make a fix after the release - it is slightly different.

What is your Istio version and your cert manager version?

**Comment by berndverst** [comment]
You should not need to change anything in the manifests file if you create a new AKS cluster.

I just updated the documentation (PR still pending) to also make it clear that no other Istio installation can coexist. Please give it a try with the new instructions once the change has been merged an is live at kubeflow.org. 
https://github.com/kubeflow/website/pull/2413

And you may want to wait another day or so for this to be merged if you care about using KFServing:
https://github.com/kubeflow/manifests/pull/1698

**Comment by berndverst** [comment]
One thing to double-check -- where is `cert-manager` installed?

It needs to be in the `cert-manager` namespace and also one related component in the `kube-system` namespace.

**Comment by uipo78** [comment]
sorry for responding so late.

* yes, istio was preinstalled following [the directions that kubeflow points to](https://docs.microsoft.com/en-us/azure/aks/servicemesh-istio-install?pivots=client-operating-system-linux). we deployed 1.7.3 at the time.
* cert manager 0.15.2 is installed in the `cert-manager` namespace.

***

> You should not need to change anything in the manifests file if you create a new AKS cluster.

we had to remove the cert manager components because kubeflow blew away our existing installation.

***

i'll see if i can get a functional installation following the changes from your PR and report back.
